### PR TITLE
Strengthen TypeScript CI and release gates

### LINKED ISSUE #22: Strengthen TypeScript CI and release gates
## Context

Recent `crap-typescript` CI/release issues apply to `cognitive-typescript` because the sibling projects should have comparable validation strength for supported runtime versions and platform-sensitive paths.

Related `crap-typescript` issues:

- fabian-barney/crap-typescript#125
- fabian-barney/crap-typescript#126
- fabian-barney/crap-typescript#127
- fabian-barney/crap-typescript#131

## Task

Strengthen CI and release workflows without adding coverage collection or CRAP-specific behavior.

CI requirements:

- Test the oldest supported Node version from `package.json` engines. If engines remain `>=18`, CI must include Node `18`.
- Also test active LTS/current Node versions used by the sibling projects; at minimum include Node `22` and `24`, and add the current release line if the project supports it.
- Add Windows coverage for platform-sensitive behavior, especially report path validation, file selection, changed-file detection, and filesystem case handling.
- Keep Linux coverage for the standard build/test/publish validation path.
- Reduce duplicate `npm ci` work where practical without weakening job isolation; use a clear matrix or dependency cache strategy.

Release requirements:

- `release.yml` must run the `cognitive-typescript` gate and the sibling `crap-typescript` gate before publishing.
- Release validation must fail before publication if either gate fails.
- Release workflow must use the same documented Node/npm compatibility assumptions as CI.

## Non-goals

- Do not force coverage, run coverage reporters, parse Istanbul/LCOV, or add CRAP score computation.
- Do not remove useful job isolation just to avoid repeated installs.

## Acceptance Criteria

- CI matrix covers Node `18` if still supported, plus active LTS/current Node versions used by the repo.
- CI has Windows coverage for path-sensitive code.
- Release workflow runs cognitive and CRAP gates before npm publish.
- Workflow dependency installation is not unnecessarily duplicated where a cache or matrix can solve it cleanly.
- README/CONTRIBUTING list the same local validation commands the workflows enforce.
- Tests/workflow checks prove report path validation and changed-file detection run on Windows.

### PR DESCRIPTION
**Pull request by fabian-barney**
Closes #22

## Summary
- add Ubuntu and Windows CI coverage for Node 22.12.0 and 24.0.0
- keep cognitive and CRAP gates explicit in CI, with isolated CRAP gate installs instead of fragile cross-job workspace handoff
- split release validation from privileged publishing and require both gates before publication

## Validation
- npm ci
- npm run build
- npm test
- npm run cognitive-typescript-check
- npm run crap-typescript-check
- npm pack --workspaces
- npm run verify-release-version -- v0.1.1

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -9,18 +9,48 @@ permissions:
...
+        run: npm install --global npm@11.14.1
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Build packages
+        run: npm run build
+
+      - name: Run tests
+        run: npm test
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/build.yml]: Valid. Fixed on `167b4cf` by removing the separate `npm run build` step from the build jobs and letting `npm test` remain the single build-plus-test entrypoint there. The local validation still passed with `npm run build`, direct Vitest, both gates, `npm pack --workspaces`, and `npm run verify-release-version -- v0.1.1`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -38,18 +68,27 @@ jobs:
...
-  cognitive-typescript-gate:
-    name: cognitive-typescript Gate
+      - name: Save crap-typescript gate workspace
+        uses: actions/cache/save@v4
+        with:
+          path: |
+            node_modules
+            packages/*/dist
+            packages/*/.tsbuildinfo
+          key: cognitive-typescript-crap-gate-ubuntu-22.12.0-${{ github.sha }}
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/build.yml]: Valid. Fixed on `167b4cf` by dropping the required cache-based full-workspace transfer entirely. The CRAP jobs now do a normal `npm ci` with the npm package cache enabled, and they consume only the prepared `dist`/`.tsbuildinfo` outputs via workflow artifacts, which is the supported same-run data-sharing mechanism for job outputs. I reran the local validation sequence after the workflow rewrite.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -11,3 +11,22 @@ All changes in this repository are expected to be issue-linked.
...
+Run the same command set the CI workflow enforces before pushing:
+
+```bash
+npm ci
+npm run build
+npm test
+npm run cognitive-typescript-check
+npm run crap-typescript-check
+npm pack --workspaces
+```
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply CONTRIBUTING.md:25]: Valid. Fixed on `167b4cf` by making the docs explicit: Ubuntu enforces the full build/test/gate/pack command set, while Windows independently validates build/test behavior for platform coverage. I updated both `CONTRIBUTING.md` and the README wording so the repository guidance matches the actual workflow split.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -24,7 +23,6 @@ jobs:
         uses: actions/setup-node@v6
         with:
           node-version: "24"
-          registry-url: "https://registry.npmjs.org"
           cache: npm
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/release.yml]: Valid. Fixed on `167b4cf` by pinning both release jobs to Node `24.0.0` so the tag workflow uses the same exact runtime version described in CI and the repository docs. I also aligned the publish job to install the pinned npm version before publishing.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -42,26 +40,63 @@ jobs:
...
+
+      - name: Run crap-typescript gate
+        run: npm run crap-typescript-check
+
+      - name: Build publish artifacts
+        run: |
+          mkdir -p release-artifacts
+          npm run build
+          npm pack --workspaces --pack-destination release-artifacts
+          cp release-notes.md release-artifacts/release-notes.md
```
[the remark this answers is not in the document]
Comment (fabian-barney) [reply .github/workflows/release.yml]: Valid. Fixed on `167b4cf` by restructuring release validation to build once up front, run Vitest directly, run the cognitive gate, run the CRAP gate via direct Vitest invocation, and then pack from the already-built outputs. That removes the repeated TypeScript rebuilds from the validation path while keeping the same gates in place.

**Comment by fabian-barney** [comment]
Superseded by #33. This keeps the exact validated head but reopens it as a fresh PR so GitHub can run a latest-head Copilot review after the earlier re-review requests produced no new review.
